### DISCUSSION TITLE
Brew package manager

### DISCUSSION THREAD

**Discussion by TuxedoHam** [hmatuschek/qdmr#236] [General]
Any chance of adding support for qdmr to being installed via Brew?

**Comment by hmatuschek** [comment]
Peter, OE1LFB worked on that and got a working brew-formula. As far as I know, it did not make it yet into the official distribution. To my knowledge, Peter hangs around at the qdmr matrix chat [#qdmr:darc.de](https://matrix.to/#/#qdmr:darc.de).

**Comment by hmatuschek** [comment]
This is something I cannot do. So I move this issue to the discussions.

**Comment by csilv** [comment]
Just plugging my wish for brew support. General recommendation is to avoid having both macports and brew installed on the same machine, and can't switch to macports as brew is required for my needs. Barring that, a binary app would included with future releases would be perfectly fine.

For now I'll attempt to built the app myself. Looking forward to using it (new user). Thank you for your efforts!

**Comment by toco** [comment]
Hannes can't submit his own tool to brew due to their [acceptable formula restrictions](https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff).
A PR to the core tap would also likely receive pushback since it builds a [.app](https://docs.brew.sh/Acceptable-Formulae#stuff-that-builds-an-app), but I guess the dmrconf tool should pass those rules.

I created a formula locally which successfully builds dmrconf and optionally the app.
Feel free to try to submit the formula to the core tap (would likely require the removal of the app bundle option) or host a custom tap for it.

`brew create https://github.com/hmatuschek/qdmr/archive/refs/tags/v0.10.2-2.tar.gz`
Paste working formula.
`brew install --build-from-source --with-app-bundle qdmr`

Formula:
```
class Qdmr < Formula
  desc "Codeplug programming tool for DMR radios"
  homepage "https://dm3mat.darc.de/qdmr/"
  url "https://github.com/hmatuschek/qdmr/archive/refs/tags/v0.10.2-2.tar.gz"
  sha256 "bac4d13ca2c489d50b67631a0031176ae111b53938c131f3d469d339a698e0fa"
  license "GPL-3.0-only"
  head "https://github.com/hmatuschek/qdmr.git", branch: "master"

  option "with-app-bundle", "Build qdmr app bundle. By default only the 'dmrconf' command line tool is built."

  depends_on "cmake" => :build
  depends_on "makeicns" => :build
  depends_on "pkg-config" => :build
  depends_on "libusb"
  depends_on "qt@5"
  depends_on "yaml-cpp"

  def install
    option_args = []

    if build.with? "app-bundle"
      option_args << "-D INSTALL_BUNDLE=YES"
      option_args << "-D BUNDLE_PATH=#{prefix}/"
    end

    system "cmake", ".", *option_args, *std_cmake_args
    system "make", "install"
    (pkgshare).install "examples"
 
    ohai "The app bundle is installed in #{prefix}." if build.with? "app-bundle"
  end

  test do
    # Custom treatment for `v0.10.2-2` since the `-2` isn't part of the embedded version number
    # We can replace it with simply `#{version}` for the next update.
    assert_match "dmrconf #{version.to_s.delete_suffix("-2")}\n", shell_output("#{bin}/dmrconf --version")
    system "#{bin}/dmrconf", "verify", "--radio", "opengd77", "#{pkgshare}/examples/kw.yaml"
    system "#{bin}/dmrconf", "verify", "--radio", "opengd77", "#{pkgshare}/examples/minimal.yaml"
  end
end
```

**Comment by rhaen** [comment]
As of now (23. August 2023) qdmr made it to the Homebrew repositories and can be installed using

`brew install qmdr`

without issues.

Tested on MacOS Venture 13.5.1 (Intel & Mac M1)

**Comment by peterloron** [comment]
Hello. I installed qdmr 0.11.3 using brew on my M1 MacBook running MacOS 14.2. The brew install went fine, but when I start the app, I get an immediate crash. The version of libyaml-cpp that is available and gets installed is 0.8.0. 

-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               qdmr [84803]
Path:                  /Users/USER/*/qdmr.app/Contents/MacOS/qdmr
Identifier:            de.darc.dm3mat.qdmr
Version:               0.10 (0.10.4)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-01-08 21:32:55.9669 -0800
OS Version:            macOS 14.2 (23C63)
Report Version:        12
Anonymous UUID:        D71A9142-91D5-8BCD-33C9-5DAC22625F90

Sleep/Wake UUID:       1F3D8F0C-7087-4198-8580-37CD59A91F7F

Time Awake Since Boot: 1200000 seconds
Time Since Wake:       354735 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: /opt/homebrew/*/libyaml-cpp.0.7.dylib
Referenced from: <F3215EF0-95B4-3187-9AA8-06799D817601> /Users/USER/*/qdmr.app/Contents/MacOS/qdmr
Reason: tried: '/opt/homebrew/*/libyaml-cpp.0.7.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/*/libyaml-cpp.0.7.dylib' (no such file), '/opt/homebrew/*/libyaml-cpp.0.7.dylib' (no such file), '/usr/local/lib/libyaml-cpp.0.7.dylib' (no such file), '/usr/lib/libyaml-cpp.0.7.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/yaml-cpp/0.8.0/lib/libyaml-cpp.0.7.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/yaml-cpp/0.8.0/lib/libyaml-cpp.0.7.dylib' (no such file), '/opt/homebrew/Cellar/yaml-cpp/0.8.0/lib/libyaml-cpp.0.7.dylib' (no such file), '/usr/local/lib/libyaml-cpp.0.7.dylib' (no such file), '/usr/lib/libyaml-cpp.0.7.dylib' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)

**Comment by skipper-henrik** [comment]
Hello all. I happily installed qdmr with homebrew because I didn't want to install complete Xcode. My Question now: Is there possibility to get a program-icon to click on? At the moment I must open terminal and type qdmr [return].

thanks for any help
